pkg/models/user_account_notification_preferences.go

```diff
@@ -0,0 +1,28 @@
+package models
+
+import (
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+)
+
+type UserAccountNotificationPreferences struct {
```

Comment:
This was a little confusing to me to know where the data is coming from and what it represents.Have you considered embedding the whole user account object and add the other property? (That way any struct drift is covered here as well. The only thing I don't love about that approach is that UserAccount comes from another package.

I think adding comments to the type and the functions would also help with the clarity of where the data comes from and what the function is doing.

```suggest
type UserAccountNotificationPreferences struct {
	authentication.UserAccount
	NewModelPlanNotification UserNotificationPreferenceFlags `db:"new_model_plan"`
}
```

Reply:
Let's talk about this tomorrow. I'd rather not invest much more time on this query, because it's downstream of some tech debt I've discussed with Clay. I'd like to replace this query entirely with something more generic by transposing the notifications table and using an ENUM field to identify notification types rather than columns. That is an intermediate / significant lift, so it needs to be handled separately to this ticket.

Reply:
Chatted about this with Tom on a call -- I think we're going to keep this as-is and look to maybe have a discussion going forward about how we pull Notification Preferences from the DB. I think it'd be nice to have / use a single `struct` type that can handle most of our cases (has all flags, has user account info), but didn't want to hold the PR just for that.

Maybe we can chat as a group sometime this week and run through the use case / design of this so we can all align on an approach!